lib/experimental/blocks.php

```diff
@@ -168,14 +171,14 @@ function gutenberg_render_block_connections( $block_content, $block, $block_inst
 				continue;
 			}
 
-			$tags  = new WP_HTML_Tag_Processor( $block_content );
-			$found = $tags->next_tag(
-				array(
-					// TODO: In the future, when blocks other than Paragraph and Image are
-					// supported, we should build the full query from CSS selector.
-					'tag_name' => $block_type->attributes[ $attribute_name ]['selector'],
-				)
-			);
+			$selectors = explode( ',', $block_type->attributes[ $attribute_name ]['selector'] );
+			$found     = false;
+			while ( ! $found && count( $selectors ) > 0 ) {
+				$tags = new WP_HTML_Tag_Processor( $block_content );
+				// TODO: In the future, to support connecting more block attributes,
+				// we should build the full query from CSS selector.
+				$found = $tags->next_tag( trim( array_shift( $selectors ) ) );
+			}
```

Comment:
we could invert this loop and avoid the confusion @talldan wants to clarify, while improving performance.

 - iterate through all the tags _until_
 - we find a tag matching one of the selectors

this assumes that there are likely more tags than selectors, many more, and so the cost of checking each selector at each tag will be cheaper than checking every tag at each selector. additionally, we should make it pretty explicit here in a comment that the _only_ selectors we currently support are tag names.

```php
$selectors = array();
foreach ( explode( ',', $block_type->attributes[ $attribute_name ]['selector'] ) as $tag_name ) {
	$selectors[] = trim( strtoupper( $tag_name ) );
}

$tags = new WP_HTML_Tag_Processor( $block_content );
while ( $tags->next_tag() ) {
	$tag_name = $tags->get_tag();
	
	if ( ! in_array( $tag_name, $selectors, true ) ) {
		 continue;
	}
	
	$markup       = "<$tag_name>$custom_value</$tag_name>";
	$updated_tags = new WP_HTML_Tag_Processor( $markup );
	$updated_tags->next_tag();

	// Get all the attributes from the original block and add them to the new markup.
	$names = $tags->get_attribute_names_with_prefix( '' );
	foreach ( $names as $name ) {
		 $updated_tags->set_attribute( $name, $tags->get_attribute( $name ) );
	}

	return $updated_tags->get_updated_html();
}
```